ext/js/data/json-schema.js

```diff
@@ -1343,12 +1343,20 @@ class JsonSchemaProxyHandler {
      * @returns {?number}
      */
     _getArrayIndex(property) {
-        if (typeof property === 'string' && this._numberPattern.test(property)) {
-            return Number.parseInt(property, 10);
-        } else if (typeof property === 'number' && Math.floor(property) === property && property >= 0) {
-            return property;
-        } else {
-            return null;
+        switch (typeof property) {
+            case 'string':
+                if (this._numberPattern.test(property)) {
+                    return Number.parseInt(property, 10);
+                }
+                break;
+            case 'number':
+                if (Math.floor(property) === property && property >= 0) {
+                    return property;
+                }
+                break;
+            default:
```

Comment:
default case not needed; omit.

Reply:
It would be preferable to have the default case if we ever turn on the switch-exhaustiveness-check rule though.

Reply:
I am not fully convinced it should be on yet, especially given that it doesn't work yet. I'm also not super fond of the idea of needing `default: break;` for things just to make a lint rule happy. We should wait until we can at least get it fully working to evaluate, and maybe only enable it as an opt-in via a meta comment.